xarray/tests/__init__.py

```diff
@@ -182,6 +182,13 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         assert_allclose(ar1, ar2, rtol=rtol, atol=atol)
 
 
+@contextmanager
+def raises_regex(error, pattern):
+    with pytest.raises(error) as excinfo:
+        yield
+    excinfo.match(pattern)
```

Comment:
I recall now that the `match` method was only added pytest 3.0. I think there's a simple work around for earlier versions of pytest, so I'm inclined to switch to that rather than require newer pytest for the test suite.